engine/component.go

```diff
@@ -0,0 +1,52 @@
+// Copyright 2021 Optakt Labs OÜ
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+package engine
+
+import (
+	"time"
+
+	"github.com/rs/zerolog"
+)
+
+// Component wraps any component that can be started and stopped.
+type Component struct {
+	log  zerolog.Logger
+	run  func() error
+	stop func()
+}
```

Comment:
Are you referring to the `Component` type? It's used in the two executables using the engine in this PR..

Reply:
I mean that in the executable its only used with the Engine.Component function not by it self. If you change to private the everything will be the same. Because it is exported but you cannot do anything with it. You have to use the Engine.Component function.

Reply:
Ah, I see what you mean. I think you're correct then, with the current code, there's no need for the Component type to be exported, since only the `Component` function is used.